sql/src/main/java/io/crate/analyze/AlterTableAddColumnAnalyzer.java

```diff
@@ -72,6 +86,12 @@ public AnalyzedAlterTableAddColumn analyze(AlterTableAddColumn<Expression> alter
         // convert and validate the column name
         ExpressionToColumnIdentVisitor.convert(tableElement.name());
 
+        // 1st phase, exclude check constraints (their expressions contain column references)
+        List<ColumnConstraint<Expression>> checkColumnConstraints = tableElement.constraints()
+            .stream()
+            .filter(CheckColumnConstraint.class::isInstance)
+            .collect(Collectors.toList());
+        tableElement.constraints().removeAll(checkColumnConstraints);
```

Comment:
Please don't mutate AST structures,  they can be cached via prepared statements and should be immutable